nixos/modules/hardware/fw-fanctrl.nix

```
@@ -0,0 +1,131 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.hardware.fw-fanctrl;
+in
+{
+  options.hardware.fw-fanctrl = {
+    enable = lib.mkEnableOption "the fw-fanctrl systemd service and install the needed packages";
+
+    disableBatteryTempCheck = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Disable checking battery temperature sensor
+      '';
+    };
+
+    config = lib.mkOption {
+      default = { };
+      description = ''
+        Config for the fw-fanctrl service (documentation: https://github.com/TamtamHero/fw-fanctrl/blob/main/doc/configuration.md)
+      '';
+      type = lib.types.submodule {
+        options = {
+          defaultStrategy = lib.mkOption {
+            type = lib.types.str;
+            default = "";
+            description = "Default strategy to use";
+          };
+
+          strategyOnDischarging = lib.mkOption {
+            type = lib.types.str;
+            default = "";
+            description = "Default strategy on discharging";
+          };
+
+          strategies = lib.mkOption {
+            default = { };
+            description = ''
+              Strategies which can be used by fw-fanctrl
+            '';
+            type = lib.types.attrsOf (
+              lib.types.submodule (
+                { name, ... }:
+                {
```

```suggestion
              lib.types.submodule {
```

looks like `name` isn't actually used anywhere